t/plugin/encode-default.t

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use utf8;
+
+use Encode qw( encode_utf8 );
+
+use Test::More;
+use Test::Exception;
+use Test::Warn;
+
+require Test::NoWarnings;
+
+use Log::Any::Plugin;
+
+use Log::Any::Test;
+use Log::Any qw( $log );
+
+
+my $msg = "鸿涛 \x{1f4A9} -- adapter with encoding set should have no warnings or errors on wide char output";
+
+
+note 'log->error expected to be available to test functionality'; {
+    $log->clear();
+
+    ok( $log->is_error, '... $log->error is enabled' );
+
+    $log->error("test");
+
+    is( scalar @{$log->msgs()}, 1, "Exactly 1 error message expected to be logged" );
+    $log->contains_ok("test", "message[test] expected to be logged");
+}
+
+note 'Encode has not been applied yet. Check default behaviour.'; {
+    $log->clear();
+
+    $log->error($msg);
+
+    $log->does_not_contain_ok(encode_utf8($msg), 'no encoded message occurs');
+    $log->contains_ok($msg, 'but still logged in original form');
+}
+
+note 'Applying Encode plugin with default encoding'; {
+    lives_ok { Log::Any::Plugin->add('Encode') }
+        'plugin applied ok';
+}
+
+note 'Check that logged message now encoded'; {
+    $log->clear();
+
+    $log->error($msg);
+
+    $log->contains_ok(encode_utf8($msg), 'message is logged and encoded');
+}
+
+
+Test::NoWarnings::had_no_warnings();
```

Comment:
If you look at the other tests you'll see I have the no-warnings tests guarded by `if $ENV{RELEASE_TESTING}`

I've been burnt in the past where some obscure platform throws some unrelated harmless warning, causing test failures, leading to installation failures.

This case is a little different, in that the intention of this module is to avoid a specific warning.

It may be worth hooking `$SIG{__WARN__}` and looking for that specific warning, and ignoring the rest.

Reply:
Hmm, I understand. Actually, I had started with `Test::Warnings` to check specifically (and that module says that works well with `Test::NoWarnings` removing the warning if checked for) -- the issue is that on the `Test` adapter, no warnings occur. In the initial SD tests, I had access to the the `Stderr` adapter, where the bug is reproducible.

Suggestions?

Reply:
Oh sorry, I was still thinking of your earlier version that used a Stderr adapter and checked that it didn't emit a warning.

For just a general 'shouldn't-have-warnings' test, just add the `if $ENV{RELEASE_TESTING}` like the other tests have.

Reply:
The Stderr adapter is part of Log::Any, so if you want to use that in a test you're welcome to.